File: leetcode-cn/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start

from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mem = {}

        for i, num in enumerate(nums):
            if target - num in mem:
                return [i, mem[target - num]]
            else:
                mem[num] = i


# @lc code=end

File: leetcode-cn/C++/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        int left = 0;
        int right = nums.size() - 1;

        unordered_map<int, int> mem;

        int i = 0;
        for (; i < nums.size(); i++) {
            if (mem.find(target - nums[i]) != mem.end()) {
                return { mem[target - nums[i]], i };
            } else {
                mem[nums[i]] = i;
            }
        }
        return {};
    }
};
// @lc code=end
